src/cogent3/evolve/likelihood_tree.py

```diff
@@ -177,10 +171,9 @@ class _PyLikelihoodTreeEdge(_LikelihoodTreeEdge):
     LOG_BASE = numpy.log(BASE)
 
     def sum_input_likelihoodsR(self, result, *likelihoods):
-        result[:] = 1.0
-        for (i, index) in enumerate(self.indexes):
-            result *= numpy.take(likelihoods[i], index, 0)
-        return result
+        return likelihood_tree.sum_input_likelihoods(
```

Comment:
See if it's possible to make the creation of `result` ensure contiguous memory. In addition, add the following
```python
if not result.flags['C_CONTIGUOUS']:
    result = numpy.ascontiguousarray(result)
```

Reply:
add the same check for `self.index`, may not be necessary but overhead very small